packages/vite/src/plugins/build.ts

```diff
@@ -56,7 +56,7 @@ export const buildPlugin = ({ config, router, manifest }: BuildPluginOpts): Plug
         build: {
           manifest: !isSsr,
           outDir: isSsr ? config.serverOutDir : config.clientOutDir,
-          target: isSsr ? 'esnext' : 'modules',
+          target: isSsr ? 'esnext' : undefined,
```

Comment:
This was the only other incompatibility I noticed. Vite < 7 defaults to modules anyways, so i just removed this and we fall back to vite's default in non ssr build.